### PR TITLE
Add Construct_closest_point_3 to Kernel

### LINKED ISSUE #260: Wrong functor required/used
The AABBGeomTraits require the functor [Compute_closest_point_3](http://doc.cgal.org/latest/AABB_tree/classAABBGeomTraits.html#a97e587bad84dcd1e90d00fb18a134762), but this signature is never used in `AABB_tree` code.

The class `AABB_traits`  provides a functor `Closest_point` which calls the free function `internal::nearest_point_3` which seems to be used instead. Note that even if this functor is renamed it is still in the wrong class (should be in the geomtraits).

### PR DESCRIPTION
**Pull request by afabri**
Add a functor Construct_closest_point_3  which before  was called Closest_point in the concept AABBTraits

This fixes #260 and #1341

### CODE REVIEW

File: Kernel_23/doc/Kernel_23/Concepts/FunctionObjectConcepts.h [hunk 1]
```diff
@@ -3887,6 +3887,54 @@ class ConstructCircumcenter_3 {
...
+
+  /**
+     * @brief Computes the closest_point from query between bound and
+     * any point of triangle.
+     * @param query the query point
+     * @param triangle the triangle
+     * @param bound the farthest point
+     * @param k the kernel
+     * @return nearest point: bound or a point inside triangle
+     */
```
Comment (janetournois) [inline Kernel_23/doc/Kernel_23/Concepts/FunctionObjectConcepts.h]: todo : add backquotes to variables names in the documentation, to get the right layout

**Comment by janetournois** [comment]
todo : it's `closest_point_3` in the implementation, and `closest_point` in the doc

**Comment by sloriot** [comment]
This should fix #1341

**Comment by sloriot** [comment]
The branch was rebased, please reset your local branch to avoid any issue

File: AABB_tree/doc/AABB_tree/Concepts/AABBTraits.h [hunk 2]
```diff
@@ -199,7 +199,7 @@ Compare_distance compare_distance_object();
 /*!
 Returns the closest point constructor. 
 */ 
-Closest_point closest_point_object(); 
+Construct_closest_point_3 construct_closest_point_object();
```
Comment (lrineau) [inline AABB_tree/doc/AABB_tree/Concepts/AABBTraits.h]: It should be `construct_closest_point_3_object();`, with a `_3`.

**Comment by sloriot** [comment]
I think we can close this PR as #1428 should replace it.

**Comment by lrineau** [comment]
@sloriot @janetournois @afabri What is the status of this PR? #1428 is currently under testing.

**Comment by afabri** [comment]
If I remember correctly we planned to cancel this PR.

**Comment by lrineau** [comment]
Replaced by #1428.

**Pull request closed without merging**